packages/nc-gui/composables/useMapViewDataStore.ts

```diff
@@ -0,0 +1,184 @@
+import { reactive } from 'vue'
+import type { ComputedRef, Ref } from 'vue'
+import type { ColumnType, MapType, PaginatedType, TableType, ViewType } from 'nocodb-sdk'
+import { IsPublicInj, ref, useInjectionState, useMetas, useProject } from '#imports'
+import type { Row } from '~/lib'
+
+const storedValue = localStorage.getItem('geodataToggleState')
+
+const initialState = storedValue ? JSON.parse(storedValue) : false
+
+export const geodataToggleState = reactive({ show: initialState })
+
+const formatData = (list: Record<string, any>[]) =>
+  list.map(
+    (row) =>
+      ({
+        row: { ...row },
+        oldRow: { ...row },
+        rowMeta: {},
+      } as Row),
+  )
+
+const [useProvideMapViewStore, useMapViewStore] = useInjectionState(
+  (
+    meta: Ref<TableType | undefined>,
+    viewMeta: Ref<ViewType | MapType | undefined> | ComputedRef<(ViewType & { id: string }) | undefined>,
+    shared = false,
+    where?: ComputedRef<string | undefined>,
+  ) => {
+    if (!meta) {
+      throw new Error('Table meta is not available')
+    }
+
+    const formattedData = ref<Row[]>([])
+
+    const { api } = useApi()
+
+    const { project } = useProject()
+
+    const { $api } = useNuxtApp()
+
+    const { isUIAllowed } = useUIPermission()
+
+    const isPublic = ref(shared) || inject(IsPublicInj, ref(false))
+
+    const { sorts, nestedFilters } = useSmartsheetStoreOrThrow()
+
+    const { fetchSharedViewData } = useSharedView()
+
+    const mapMetaData = ref<MapType>({})
+
+    const geoDataFieldColumn = ref<ColumnType | undefined>()
+
+    const defaultPageSize = 1000
+
+    const paginationData = ref<PaginatedType>({ page: 1, pageSize: defaultPageSize })
```

Comment:
im wondering how to use paginate in a Map view. can you explain it?

Reply:
So the default value of the pagination in nocodb is 25 records. However for the map view it makes sense to show all the records as long it is not causing performance problems. As a safe limit we set 1000 records to be shown on a map view for this MVP and we'll see what is the best solution for the larger data set later. (e.g. one thing Pranav mentioned is to load more records by multiple calls to the BE).